docs/content/topics/project-developers/events.md

```diff
@@ -9,4 +9,229 @@ aliases:
+[JavaScript/TypeScript SDK Event].
+
+[GitHub Gateway]: https://github.com/brigadecore/brigade-github-gateway
+[Go SDK Event]: https://github.com/brigadecore/brigade/blob/v2/sdk/v2/core/events.go
+[JavaScript/TypeScript SDK Event]: https://github.com/brigadecore/brigade-sdk-for-js/blob/master/src/core/events.ts
+
+## Event Subscriptions
+
+In order to receive events, a Brigade project must explicitly subscribe to
+them. This is done via an Event Subscription configured on the project.
+
+For an overview of a project definition in Brigade, see the [Projects] doc.
+
+Here we'll look at a sample project and dig into its `eventSubscriptions`
+section:
+
+```yaml
+apiVersion: brigade.sh/v2-beta
+kind: Project
+metadata:
+  id: hello-world
+description: Demonstrates responding to an event with brigadier
+spec:
+  eventSubscriptions:
+  - source: brigade.sh/cli
+    types:
+    - exec
+```
+
+The `hello-world` project above subscribes to one event, that of source
+`brigade.sh/cli` and type `exec`. This is the event emitted from the
+`brig event create` command.
+
+Additional events and/or additional types of a given event are added under this
+`eventSubscriptions` section. [Qualifiers] and [Labels] may also be added, via
+the `qualifiers` and `labels` fields respectively, to further refine inbound
+events.
+
+Here's a look at an event subscription configuration using all of the above:
+
+```yaml
+eventSubscriptions:
+- source: brigade.sh/github
+  qualifiers:
+    repo: brigadecore/brigade
+  labels:
+    branch: main
+  types:
+  - push
+```
+
+[Projects]: /topcs/project-developers/projects
+[Qualifiers]: #qualifiers
+[Labels]: #labels
+
+## Handling Events
+
+Events that successfully reach a subscribed project can be handled in the
+project's Brigade script. Let's revisit the `hello-world` project definition
+from above and add an inline `brigade.js` script via the `defaultConfigFiles`
+section of its Worker spec. In the script, we'll look at the event handler for
+inbound events of source `brigade.sh/cli` and type `exec`:
+
+```yaml
+apiVersion: brigade.sh/v2-beta
+kind: Project
+metadata:
+  id: hello-world
+description: Demonstrates responding to an event with brigadier
+spec:
+  eventSubscriptions:
+  - source: brigade.sh/cli
+    types:
+    - exec
+  workerTemplate:
+    defaultConfigFiles:
+      brigade.js: |
+        const { events } = require("@brigadecore/brigadier");
+
+        events.on("brigade.sh/cli", "exec", async event => {
+          console.log("Hello, World!");
+        });
+
+        events.process();
+```
+
+The following section of Javascript comprises the event handler:
+
+```javascript
+events.on("brigade.sh/cli", "exec", async event => {
+  console.log("Hello, World!");
+});
+```
+
+In fact, all event handlers follow the same pattern, that of:
+
+```javascript
+events.on("event source", "event type", async event => {
+  // handle event
+});
+```
+
```

Comment:
Is it worth mentioning that scripts don't _need_ to be embedded in project definitions and that this is merely one option that happens to be suitable for a simple demo?

I know some people could be instantly turned off at the thought of having to embed JS in YAML.